pkgs/development/web/insomnium/default.nix

```diff
@@ -0,0 +1,80 @@
+{ pkgs, stdenv, lib, makeDesktopItem, makeWrapper, fetchurl, appimageTools
+, undmg, ... }:
+let
+  pname = "insomnium";
+  version = "0.2.3-a";
+  appname = "Insomnium";
+  name = "${appname}.Core-${version}";
+  meta = with lib; {
+    description =
+      "Fast local API testing tool that is privacy-focused and 100% local";
+    homepage = "https://github.com/ArchGPT/insomnium";
+    license = licenses.mit;
+    maintainers = [ "nephalemsec" ];
+  };
+
+  filename =
+    if stdenv.isDarwin then "${name}.signed.dmg" else "${name}.AppImage";
+
+  src = pkgs.fetchurl {
+    url =
+      "https://github.com/ArchGPT/insomnium/releases/download/core%40${version}/${filename}";
+    sha256 = if stdenv.isDarwin then
+      "sha256-OlYfoNNBPSMYDVSIsANKW7yy1DPkYA4x0ALgyipS2d8="
+    else
```

Comment:
I don't think it's fair to assume that `!stdenv.isDarwin` is `stdenv.Linux && stdenv.x86_64` (or even `stdenv.isLinux`). I'd prefer a proper distinction as in the following:

https://github.com/NixOS/nixpkgs/blob/782e77866f7c36de4e208b04894f2ed4f6ad8fb0/pkgs/applications/science/biology/blast/bin.nix#L18-L33